### PR TITLE
feat: CHECKS ITERATION PARAMETERS WHEN GETTING LLCMODEL DATASETS

### PR DESCRIPTION
**Pull request by fraserwg**
When loading datasets using the llcreader the user is able to specify the iteration start, stop and step. These can also be specified as attributes in the `BaseLLCModel`. This commit introduces a check that compares the `iter_start` and `iter_step` fed into `get_dataset` with the `BaseLLCModel` attributes. If they are found to be incompatible a warning is triggered.

A warning over an Error was chosen as subsuquent code may still execute successfully even if there is a mismatch between the user specified and `BaseLLCModel` derived values for `iter_start` and `iter_stop`.

### CODE REVIEW

**Comment by fraserwg** [comment]
This implements my request from issue #235

**Comment by rabernat** [comment]
Thanks a lot for this! We really appreciate your willingness to dive into xmitgcm.

We strive for high test coverage for this package. Therefore, you will need to add a test for this new feature. You will probably want to use the pytest warnings mechanism: https://docs.pytest.org/en/3.0.5/recwarn.html.

**Comment by rabernat** [comment]
It looks like there is now a merge conflict here, since we recently merged #231. Perhaps @timothyas can help you sort it out?

**Comment by timothyas** [comment]
No problem, it seems like the merge conflicts are all gone - thanks @fraserwg! I think this looks good to me, I just have one minor request to add to the tests, but then I think it's good to go.

**Review by timothyas — CHANGES_REQUESTED** [review]
Could you add a test which checks for this warning while calling `get_dataset`? For example at the bottom of this test:
https://github.com/MITgcm/xmitgcm/blob/04ec9fb7a819181545dcb455f8553bb5695bbcf6/xmitgcm/test/test_llcreader.py#L127
just another `with pytest.warns` check as you did for the others

File: xmitgcm/test/test_llcreader.py [hunk 1]
```diff
@@ -165,6 +166,25 @@ def test_ecco_portal_latlon(ecco_portal_model):
             assert len(ds_ll[fld].data.chunks)==1
             assert (len(ds_ll[fld]),)==ds_ll[fld].data.chunks[0]
 
+
+def test_iteration_start_check(ecco_portal_model):
+    with pytest.warns(RuntimeWarning) as record:
+        ecco_portal_model._check_iter_start(ecco_portal_model.iter_start - 1)  # Should give runtime warning
+        ecco_portal_model._check_iter_start(ecco_portal_model.iter_start)  # Should give no warning
+        ecco_portal_model._check_iter_start(ecco_portal_model.iter_start + ecco_portal_model.iter_step) # Should give no warning
```
Comment (rabernat) [inline xmitgcm/test/test_llcreader.py:174]: This is not quite what I had in mind for a test. You should not call private functions (e.g. `._check_iter_start`) from the test suite. Instead, you should call the public API (e.g. `get_dataset`) as suggested by @timothyas.

**Review by rabernat — CHANGES_REQUESTED** [review]
Thanks for your work on this @fraserwg and for your patience as we review your PR.

File: xmitgcm/test/test_llcreader.py (hunk 1 above)
Comment (fraserwg) [reply xmitgcm/test/test_llcreader.py:174]: Sure, I've rewritten the tests to call `get_dataset` but the ECCO Data Portal is currently down. I'll push the changes when I've verified they work locally 👍

**Comment by rabernat** [comment]
I think the test failure is a fluke, so I'm going to merge this now. Thanks for another great contribution!

**Pull request merged**